### ISSUE TITLE
karafka 2.0-wip does not work on read-only filesystem

### ISSUE THREAD

**Issue by ojab** [karafka/karafka#661]
During [setup](https://github.com/karafka/karafka/blob/2.0-wip/lib/karafka/setup/config.rb#L27) We're trying to create logger instance, which tries to open a file before any configuration (i. e. logging to STDOUT) takes place:
```
irb(main):001:0> require 'karafka'
/usr/local/bundle/bundler/gems/karafka-e3bc8d746594/lib/karafka/instrumentation/logger.rb:55:in `initialize': Read-only file system @ rb_sysopen - /app/log/development.log (Errno::EROFS)
	from /usr/local/bundle/bundler/gems/karafka-e3bc8d746594/lib/karafka/instrumentation/logger.rb:55:in `open'
	from /usr/local/bundle/bundler/gems/karafka-e3bc8d746594/lib/karafka/instrumentation/logger.rb:55:in `file'
	from /usr/local/bundle/bundler/gems/karafka-e3bc8d746594/lib/karafka/instrumentation/logger.rb:37:in `target'
	from /usr/local/bundle/bundler/gems/karafka-e3bc8d746594/lib/karafka/instrumentation/logger.rb:25:in `initialize'
	from /usr/local/bundle/bundler/gems/karafka-e3bc8d746594/lib/karafka/setup/config.rb:27:in `new'
	from /usr/local/bundle/bundler/gems/karafka-e3bc8d746594/lib/karafka/setup/config.rb:27:in `<class:Config>'
	from /usr/local/bundle/bundler/gems/karafka-e3bc8d746594/lib/karafka/setup/config.rb:14:in `<module:Setup>'
	from /usr/local/bundle/bundler/gems/karafka-e3bc8d746594/lib/karafka/setup/config.rb:6:in `<module:Karafka>'
	from /usr/local/bundle/bundler/gems/karafka-e3bc8d746594/lib/karafka/setup/config.rb:3:in `<top (required)>'
	from /usr/local/bundle/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:26:in `require'
	from /usr/local/bundle/gems/zeitwerk-2.4.2/lib/zeitwerk/kernel.rb:26:in `require'
	from /usr/local/bundle/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:409:in `const_get'
	from /usr/local/bundle/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:409:in `block (2 levels) in eager_load'
	from /usr/local/bundle/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:733:in `block in ls'
	from /usr/local/bundle/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:725:in `foreach'
	from /usr/local/bundle/gems/zeitwerk-2.4.2/lib/zeitwerk/loader.rb:725:in `ls'
	... 25 levels...
```

## Expected behavior

Karafka application shouldn't use log until configured & needed. 

## Actual behavior

boom!

## Steps to reproduce the problem

`chmod -rw log/development.log && ruby -rkarafka`

## Your setup details
`bundle info`, since `karafka info` doesn't work currently.
```
  * karafka (2.0.0.pre.alpha1 e3bc8d7)
	Summary: Ruby based framework for working with Apache Kafka
	Homepage: https://github.com/karafka/karafka
	Path: /home/ojab/.rbenv/versions/3.0.0/lib/ruby/gems/3.0.0/bundler/gems/karafka-e3bc8d746594
```

**Comment by ojab** [comment]
Not sure if I even should report bug to 2.0-wip, but setting up a new service with ruby-3:
- waterdrop-1.4 doesn't work with ruby-3
- karafka-1.4 doesn't work with waterdrop-2.0
/shrug

Said above, is waterdrop-1.4 alive enough to make a PR with ruby-3 fixes and what is the preferred path here from your POV? Just downgrade ruby for the time being, try to use `2.0-wip`, or something else entirely?

Maybe you can cut karafka-1.9.9.pre with producers/waterdrop removed from the codebase, so current stable karafka can be used with waterdrop-2.0?

**Comment by mensfeld** [comment]
Hey @ojab thanks for trying the 2.0 wip out :) it's still a wip so I cannot give you any quality warranties on it :(

Waterdrop 1.4 is alive enough to make Ruby 3.0 fixes if you are up to the task :) It is fully maintained, same as Karafka 1.4. I just haven't had enough time to tackle both versions the same time.

Running Ruby 3.0.0 is in general not my style :D (especially for production). I would wait for at least 3.0.1 - that comes from years of my experience :D

If you need the Karafka 1.4 without waterdrop I can cut you a release of that nature. Karafka 2.0 will not be dependant on waterdrop - it will detect its presence and configure itself but will run without it as well.

**Comment by mensfeld** [comment]
@nijikon would you mind switching waterdrop 1.4 to use Github Actions instead of Travis? The scope of changes is small and I can quickly roll out 3.0 support in 1.4.

**Comment by mensfeld** [comment]
I started a branch for 3.0 support for 1.4 WaterDrop: https://github.com/karafka/waterdrop/pull/174

**Comment by mensfeld** [comment]
@ojab 3.0 support for 1.4 is done. It's released with newest 1.4.x.

This issue has been (for 2.0) tackled here: https://github.com/karafka/karafka/pull/659
